src/NuGet.LibraryModel/LibraryDependencyTypeFlag.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+
+namespace NuGet.LibraryModel
+{
+    public class LibraryDependencyTypeFlag
```

Comment:
Since the flag is used in a HashSet, should it implement `IEquatable<T>` interface? User can create flags arbitrarily.